caddyhttp/maxrequestbody/maxrequestbody.go

```diff
@@ -0,0 +1,152 @@
+package maxrequestbody
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/mholt/caddy"
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+const (
+	serverType = "http"
+	pluginName = "maxrequestbody"
+)
+
+func init() {
+	caddy.RegisterPlugin(pluginName, caddy.Plugin{
+		ServerType: serverType,
+		Action:     setupMaxRequestBody,
+	})
+}
+
+func setupMaxRequestBody(c *caddy.Controller) error {
+	config := httpserver.GetConfig(c)
+
+	if !c.Next() {
+		return c.ArgErr()
+	}
+
+	args := c.RemainingArgs()
+	pathLimit, err := parseArguments(args)
+	if err != nil {
+		return c.ArgErr()
+	}
+
+	SortPathLimits(pathLimit)
+
+	config.MaxRequestBodySizes = pathLimit
+	fmt.Println(pathLimit)
+
+	return nil
+}
+
+func parseArguments(args []string) ([]httpserver.PathLimit, error) {
+	pathLimit := []httpserver.PathLimit{}
+	// Arguments appear in pairs of "<path> <limit>"
+	// If there's an argument that's not paired, it's assumed to be
+	// the limit used for all remaining paths ("/")
+	if len(args)%2 == 1 {
```

Comment:
Well I think everything should have a pair, with the only exception being the first argument.

The 2 cases you mentioned would fail due to the ParseInt below, also I have no idea how to interpret them.